github/compliance/issue.py

```diff
@@ -226,6 +229,44 @@ def _update_issue(
     return issue
 
 
+def _update_problems_comment(
+    issue: github3.github.issues.ShortIssue,
+    message: str | None = None,
+):
+    '''
+    Create or update a comment on an issue stating encountered problems if there are any.
+    Delete existing comment if the problems are resolved.
+    '''
+
+    issue_problems_comment_header = (
+        'An error occurred when updating this issue:'
+    )
+
+    def _problems_comment() -> github3.issues.comment.IssueComment:
+        for comment in issue.comments():
+            if comment.body.strip().startswith(issue_problems_comment_header):
+                return comment
+
+    problems_comment = _problems_comment()
+
+    if message:
+        job_url = concourse.util.own_running_build_url()
+        comment_body = f'{issue_problems_comment_header}\n\n{message}\nJob url: {job_url}'
+        if not problems_comment:
+            # Create new comment containing the message
+            issue.create_comment(comment_body)
+        else:
+            # Update comment
+            problems_comment.edit(comment_body)
+    else:
```

Comment:
so this same function will remove an existing comment, if it is called w/o a message to report? hmmm..

this (cleanup) will have the downside of us having to issue at least one additional gh-api-request for each issue, will it not? maybe we should rather omit the cleanup functionality, and accept outdated comments to be persistent.

we might add (upon successful issue-updates) a timestamp indicating the date of last update. which would be enough for users to understand an error-indicating comment to be outdated.